test/lsptoolshost/integrationTests/fileBasedPrograms.integration.test.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+import {
+    activateCSharpExtension,
+    closeAllEditorsAsync,
+    getCompletionsAsync,
+    openFileInWorkspaceAsync,
+    revertActiveFile,
+    sleep,
+    waitForAllAsyncOperationsAsync,
+    waitForExpectedResult,
+} from './integrationHelpers';
+import { describe, beforeAll, beforeEach, afterAll, test, expect, afterEach } from '@jest/globals';
+import { CSharpExtensionExports } from '../../../src/csharpExtensionExports';
+
+const doRunSuite = process.env['ROSLYN_SKIP_TEST_FILE_BASED_PROGRAMS'] !== 'true';
+console.log(`process.env.ROSLYN_SKIP_TEST_FILE_BASED_PROGRAMS: ${process.env.ROSLYN_SKIP_TEST_FILE_BASED_PROGRAMS}`);
+console.log(`doRunSuite: ${doRunSuite}`);
```

Comment:
consider extracting all this to a helper function similar to https://github.com/dotnet/vscode-csharp/blob/65e5e0872fc0eeffb9c5a5ec68183fd8fe192edb/test/lsptoolshost/integrationTests/integrationHelpers.ts#L272